### DISCUSSION TITLE
best way to upgrade Debian 11 to Debian 12

### DISCUSSION THREAD

**Discussion by lhw-sa** [uyuni-project/uyuni#7526] [Q&A]
Hi, what is the best way to upgrade from Debian 11 to Debian 12 using Uyuni?

**Comment by digdilem-work** [comment]
*   Hi, what is the best way to upgrade from Debian 11 to Debian 12 using Uyuni?

This is untested, but perhaps something like:

  1.  Upgrade machine fully under 11.
  2.  Deploy a new /etc/apt/sources.list with the repos changes to Debian 12 (and/or change the software group in Uyuni and highstate it if you’re using Uyuni as a repo)
  3.  Run the usual dist-upgrade procedures

Obviously that needs a little fine tuning and testing, but it’s how I would approach the problem.

**Comment by lhw-sa** [comment]
Thank you, that worked.
I changed the software group in Uyuni and applied the change. 
```
apt update 
apt full-upgrade
```
did it. After Reboot, /etc/debian_version shows Debian 12.
